ansible_collections/arista/avd/plugins/filter/convert_dicts.py

```diff
@@ -43,38 +42,44 @@ def convert_dicts(dictionary, primary_key="name", secondary_key="items"):
     Parameters
     ----------
     dictionary : any
-        Nested Dictionary to convert - returned untouched if not a nested dictionary
+        Nested Dictionary to convert - returned untouched if not a nested dictionary and list
     primary_key : str, optional
         Name of primary key used when inserting outer dictionary keys into items.
     secondary_key : str, optional
-        Name of secondary key used when inserting dictionary values which are list of dictionaries into items.
+        Name of secondary key used when inserting dictionary values which are list into items.
 
     Returns
     -------
     any
-        Returns list of dictionaries or input variable untouched if not a nested dictionary/list of dictionaries
+        Returns list of dictionaries or input variable untouched if not a nested dictionary/list.
     """
-    if not isinstance(dictionary, dict) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
+    if not isinstance(dictionary, dict) and not isinstance(dictionary, list) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
         # Not a dictionary, return the original
         return dictionary
+    elif isinstance(dictionary, list):
+        output = []
+        for element in dictionary:
+            if not isinstance(element, dict):
+                item = {}
+                item.update({primary_key: element})
+                output.append(item)
+            else:
+                output.append(element)
+        return output
     else:
         output = []
         for key in dictionary:
             if dictionary[key] is None:
                 # Catch cornercase where dictionary has no value because of old data models
                 output.append({primary_key: key})
             elif not isinstance(dictionary[key], dict):
-                if isinstance(dictionary[key], list):
-                    # Not a nested dictionary but a list of dictionaries, add secondary key for the list
+                # Not a nested dictionary, add secondary key for the list elements if secondary key is provided
+                if secondary_key is not None:
                     item = {}
-                    secondary_element = []
                     item.update({primary_key: key})
-                    for element in dictionary[key]:
-                        secondary_element.append(element)
-                    item.update({secondary_key: secondary_element})
+                    item.update({secondary_key: dictionary[key]})
                     output.append(item)
                 else:
-                    # Not a nested dictionary but a string, return the original
                     return dictionary
```

Comment:
I am not sure about this part - it seems to assume that all the elements are the same inside the dictionary (which may be true)

Reply:
yes, this is how it works as of now.

Reply:
Since we cannot fix the entry without having the secondary_key defined (case above) we just give up. It would not help us to do a partial convertion. Shivani had an error message here earlier, but I was not sure if we could give a meaningful error message, since we don't know the name of the outer var.
In general this case "should not happen" so either we can raise an exception, or we can ignore silently.

Reply:
ok - understood I will resolve this